hubblestack/extmods/modules/audit.py

```diff
@@ -1,450 +1,544 @@
-        if __salt__['match.compound'](match):
-            ret.extend(data)
+    Returns:
+        list -- List of audit files
+    """
+    if not audit_files:
+        log.warning('nova.audit called without any audit_files')
+        return None
 
-    return ret
+    if not isinstance(audit_files, list):
+        audit_files = audit_files.split(',')
 
+    # prepare paths
+    return ['salt://' + BASE_DIR_NOVA_PROFILES + os.sep + audit_file.replace('.', os.sep) + '.yaml'
+                    for audit_file in audit_files]
+    
 
-def _version_cmp(version):
+def __evaluate_results(result_dict, combined_dict, show_compliance):
     """
-    Handle version comparison for audit checks
-
-    :param version:
-        Version comparison string. See module-level documentation for more
-        details.
+    Evaluate the result dictionary to be returned by the audit module
+    :param result_dict:
+    :param combined_dict:
+    :param show_compliance:
     :return:
-        Boolean as to whether the versions match
     """
-    # '>=3.0.0,<3.1.0;>=3.2.0'
-    versions = version.split(';')
-    # ['>=3.0.0,<3.1.0', '>=3.2.0']
-    versions = [item.split(',') for item in versions]
-    # [['>=3.0.0', '<3.1.0'], ['>=3.2.0']]
-    processed_versions = []
-    for item in versions:
-        # Inner matches (comma separator) are AND
-        overall_match = True
-        for comparison in item:
-            if not comparison:
-                match = False
-            elif comparison.startswith('<='):
-                comparison = comparison[2:]
-                match = StrictVersion(__grains__['hubble_version']) <= StrictVersion(comparison)
-            elif comparison.startswith('<'):
-                comparison = comparison[1:]
-                match = StrictVersion(__grains__['hubble_version']) < StrictVersion(comparison)
-            elif comparison.startswith('>='):
-                comparison = comparison[2:]
-                match = StrictVersion(__grains__['hubble_version']) >= StrictVersion(comparison)
-            elif comparison.startswith('>'):
-                comparison = comparison[1:]
-                match = StrictVersion(__grains__['hubble_version']) > StrictVersion(comparison)
-            else:  # Equals, by default
-                match = StrictVersion(__grains__['hubble_version']) == StrictVersion(comparison)
-            # If we ever get a False, this whole AND block will be marked as False
-            overall_match = overall_match and match
-        processed_versions.append(overall_match)
-
-    # Outer matches (semicolon separator) are OR
-    version_match = False
-    for item in processed_versions:
-        if item:
-            version_match = True
-
-    return version_match
-
-
-def _run_audit(ret, audit_data, tags, labels, audit_file):
+    for audit_file in combined_dict:
+        result_list = combined_dict[audit_file]
+        for result in result_list:
+            sub_check = result.get('sub_check', False)
```

Comment:
You are ignoring results after this, this is needed during boolean expression evaluation later.

Reply:
Boolean expressions will be evaluated in this code here. The logic needs to be modified here only when boolean expression is going to be written.